NineChronicles.Headless/GraphTypes/StateQuery.cs

```diff
@@ -346,7 +349,136 @@ public StateQuery()
+                    //Enemy
+                    var enemyArenaAvatarStateAdr = ArenaAvatarState.DeriveAddress(enemyAvatarAddress);
+                    var enemyArenaAvatarState = context.Source.GetArenaAvatarState(enemyArenaAvatarStateAdr, enemyAvatar);
+                    List<Guid> enemyArenaEquipementList = enemyAvatar.inventory.Equipments.Where(f=>enemyArenaAvatarState.Equipments.Contains(f.ItemId)).Select(n => n.ItemId).ToList();
+                    List<Guid> enemyArenaCostumeList = enemyAvatar.inventory.Costumes.Where(f=>enemyArenaAvatarState.Costumes.Contains(f.ItemId)).Select(n => n.ItemId).ToList();
+
+                    var enemyRuneSlotStateAddress = RuneSlotState.DeriveAddress(enemyAvatarAddress, BattleType.Arena);
+                    var enemyRuneSlotState = context.Source.TryGetState(enemyRuneSlotStateAddress, out List enemyRawRuneSlotState)
+                        ? new RuneSlotState(enemyRawRuneSlotState)
+                        : new RuneSlotState(BattleType.Arena);
+
+                    var enemyRuneStates = new List<RuneState>();
+                    var enemyRuneSlotInfos = enemyRuneSlotState.GetEquippedRuneSlotInfos();
+                    foreach (var address in enemyRuneSlotInfos.Select(info => RuneState.DeriveAddress(enemyAvatarAddress, info.RuneId)))
+                    {
+                        if (context.Source.TryGetState(address, out List rawRuneState))
+                        {
+                            enemyRuneStates.Add(new RuneState(rawRuneState));
+                        }
+                    }
+
+                    var myArenaPlayerDigest = new ArenaPlayerDigest(
+                        myAvatar,
+                        myArenaEquipementList,
+                        myArenaCostumeList,
+                        myRuneStates);
 
+                    var enemyArenaPlayerDigest = new ArenaPlayerDigest(
+                        enemyAvatar,
+                        enemyArenaEquipementList,
+                        enemyArenaCostumeList,
+                        enemyRuneStates);
+
+                    Random rnd  =new Random();          
+
+                    int win = 0;
+                    int loss = 0;
+
+                    for (var i = 0; i < 1000; i++)
+                    {
+                        LocalRandom iRandom = new LocalRandom(rnd.Next());
+                        var simulator = new ArenaSimulator(iRandom);
+                        var log = simulator.Simulate(
+                            myArenaPlayerDigest,
+                            enemyArenaPlayerDigest,
+                            arenaSheets);
+                        if(log.Result.ToString() == "Win")
+                        {
+                            win++;
+                        }
+                        else
+                        {
+                            loss++;
+                        }        
+                    }
+                    return Math.Round(((decimal)win / 1000) * 100m, 2);
```

Comment:
Returning aggregated summary makes sense but I also think that more detailed information might be more helpful.

e.g.,

```json
{
    "blockIndex": 1,
    "result": [
        {
            "seed": 1234,
            "win": true
        },
        {
            "seed": 5678,
            "win": false
        },
    ]
}
```

Reply:
Thanks Swen.

So BlockIndex,
Percentage and then a 
Results List per seed/win state?

Then user can decide if they want the result list?

Reply:
@longfin 
Like so?
`{
  "data": {
    "simulationQuery": {
      "arenaPercentageCalculator": {
        "blockIndex": 7133512,
        "result": [
          {
            "seed": 1893608300,
            "win": true
          },
          {
            "seed": 2073532131,
            "win": true
          },
          {
            "seed": 1826085108,
            "win": false
          },
          {
            "seed": 1985649038,
            "win": false
          },
          {
            "seed": 614892139,
            "win": true
          },
          {
            "seed": 2060699251,
            "win": false
          },
          {
            "seed": 784343549,
            "win": true
          },
          {
            "seed": 2109394172,
            "win": true
          },
          {
            "seed": 1339463095,
            "win": true
          },
          {
            "seed": 1834517228,
            "win": false
          }
        ],
        "winPercentage": 60
      }
    }
  },
  "extensions": {}
}`

Reply:
@FioX0 right 😄